src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -193,15 +223,10 @@ private boolean isMatchingByContextStrategy(TreeWalkerAuditEvent event) {
             else {
                 eventAst = getEventAst(event);
             }
-            final Set<Integer> childAstLineNoList = getChildAstLineNo(eventAst);
-            final int min = Collections.min(childAstLineNoList);
-            final int max = Collections.max(childAstLineNoList);
-            for (int currentLine = min; currentLine <= max; currentLine++) {
-                result = lineMatching(currentLine);
-                if (result) {
-                    break;
-                }
-            }
+            final Map<String, Integer> childAstLineNoMap = getChildAstLineNo(eventAst);
```

Comment:
Is it possible that child AST has lower line number than its parent? Probably no, eventAst's line is the first line number in the lines range, so it's minimum. You need to find only max line.

Reply:
@baratali it is possible for this situation:

```
|--VARIABLE_DEF -> VARIABLE_DEF [8:4]
    |   |--MODIFIERS -> MODIFIERS [8:4]
    |   |--TYPE -> TYPE [8:4]
    |   |   |--SINGLE_LINE_COMMENT -> // [7:12]
    |   |   |   `--COMMENT_CONTENT ->  violation without filter\n [7:14]
    |   |   `--LITERAL_INT -> int [8:4]
    |   |--IDENT -> c [8:8]
    |   `--SEMI -> ; [8:9]
```

Reply:
ok, I wanted to remove complex structure `Map<String, Integer>`, but let's keep it then